scripts/pre_commit_linter.py

```diff
@@ -2178,6 +2192,30 @@ def _check_codeowner_file(self):
                             print ('WARNING! %s/%s is not covered under '
                                    'CODEOWNERS' % (root, file_name))
 
+            # Checks that the most important rules/patterns are at the bottom
+            # of the CODEOWNERS file.
+
+            # This list is populated by the boolean values resulted from
+            # matching the last lines of the CODEOWNERS file with the list of
+            # the important CODEOWNERS paths.
+            important_path_match_bool_list = ([
+                imp_path in path_lines[-1 * len(
```

Comment:
Is someone adds a new pattern(important) in the top of the list, will this check ask users to add this list over here or will this check fail?

Reply:
No, the check will fail. The 'n'-list and the last 'n' lines need to be in sync.

Reply:
![image](https://user-images.githubusercontent.com/16653571/56900733-293c8380-6ab4-11e9-94c1-2c4535b3e3bf.png)
I tried adding a new file in the CODEOWNERS critical section. (without doing any changes in pre_commit_linter.py)

What should be the expected behavior?
I got this:
![image](https://user-images.githubusercontent.com/16653571/56900805-5426d780-6ab4-11e9-986b-4ff7e45bf37d.png)

Reply:
This is happening because of the new important rules that are not in sync with the `CODEOWNER_IMPORTANT_PATHS` list. The (outdated) list checks for the last n-lines of the CODEOWNERS file in which the erring pattern is not found and hence the error.

Reply:
I'm happy that it's failing but the error message is something unrelated. Can we do something for that?

Reply:
I have modified the error message. PTAL!
Thanks.